bazel-mode.el

```diff
@@ -21,6 +21,36 @@
 ;;
 ;;; Code:
 
+(defgroup bazel-mode nil
+  "Major mode for Bazel BUILD files."
+  :link '(url-link "https://github.com/bazelbuild/emacs-bazel-mode")
+  :group 'languages)
+
+(defcustom buildifier-cmd "buildifier"
+  "Filename of buildifier executable."
+  :type 'file
+  :group 'bazel-mode)
+
+(defun bazel-mode--buildifier ()
+  "Format current buffer using buildifier."
+  (interactive)
+  (let ((build-file-contents (buffer-string))
+        (input-buffer (current-buffer))
+		(buildifier-buffer (get-buffer-create "*buildifier*")))
+    (with-current-buffer buildifier-buffer
+      (erase-buffer)
+      (insert build-file-contents)
+      (let ((return-code
+             (call-process-region (point-min) (point-max) buildifier-cmd t t)))
```

Comment:
`process-file-region` indeed doesn't exist.
Then you should probably call `process-file` instead to make this work with remote files. You'll need to write out the standard input into a temporary file.

Reply:
I'm not sure I understand this request. Is it to support the case when the BUILD file is remote? Currently we read its contents into the *buildifier* buffer and run buildifier on that. If the command succeeds, that buffers contents are copied back into the BUILD file's buffer (would this fail for a remote BUILD file?) and close the scratch buffer. If the command fails, its contents are the error message and we display it. What fails in this process if the BUILD file is remote, and how will using a temp file instead of a scratch buffer allow us to do differently?

Reply:
This is for the case where the buildifier program itself is remote. It might not be installed locally, or an incorrect/outdated version might be installed, thus if the file is remote, then the remote buildifier program should also be run.